### PR TITLE
Download external resources preference

### PR DESCRIPTION
**Pull request by angelozerr**
Download external resources preference

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: src/settings/settings.ts [hunk 1]
```diff
@@ -226,8 +226,10 @@ export function getXMLSettings(javaHome: string | undefined, logfile: string, ex
   const isWorkspaceTrusted = (workspace as any).isTrusted;
   if (isWorkspaceTrusted !== undefined && !isWorkspaceTrusted) {
```
Comment (rgrunber) [inline src/settings/settings.ts:227]: If you insert a breakpoint here, you'll see there are 2 instances of this settings :

`xml.preferences.downloadExternalRessource`
`xml.downloadExternalResource`

Just make sure whichever namespace is chosen, it is the one the language server is able to parse.

Comment (angelozerr) [reply src/settings/settings.ts:227]: I change and fix lemminx with `xml.downloadExternalResources`

File: src/settings/settings.ts [hunk 2]
```diff
@@ -225,9 +225,11 @@ export function getXMLSettings(javaHome: string | undefined, logfile: string, ex
   // Check workspace trust
   const isWorkspaceTrusted = (workspace as any).isTrusted;
   if (isWorkspaceTrusted !== undefined && !isWorkspaceTrusted) {
-    xml['xml']['validation']['resolveExternalEntities'] = false;
+    xml['xml']['validation']['resolveExternalEntities']['enabled'] = false;
+    xml['xml']['downloadExternalResources'] = false;
```
Comment (fbricon) [inline src/settings/settings.ts:229]: xml['xml']['downloadExternalResources']['enabled'] = false;

Comment (angelozerr) [reply src/settings/settings.ts]: fixed

File: src/settings/settings.ts [hunk 3]
```diff
@@ -225,9 +225,11 @@ export function getXMLSettings(javaHome: string | undefined, logfile: string, ex
   // Check workspace trust
   const isWorkspaceTrusted = (workspace as any).isTrusted;
   if (isWorkspaceTrusted !== undefined && !isWorkspaceTrusted) {
-    xml['xml']['validation']['resolveExternalEntities'] = false;
+    xml['xml']['validation']['resolveExternalEntities']['enabled'] = false;
```
Comment (rgrunber) [inline src/settings/settings.ts:228]: You would need to also update the settings in `package.json` and in the LemMinX settings structure.

Currently I'm seeing : `Activating extension 'redhat.vscode-xml' failed: Cannot create property 'enabled' on boolean 'false'.`

Comment (angelozerr) [reply src/settings/settings.ts]: Sorry for this mistake. The correct settings is `xml.resolveExternalEntities`. I fixedit.

**Comment by rgrunber** [comment]
One small case I noticed :

1) Open a workspace and do not grant trust. Remote schemas won't download, as expected.
2) Now grant workspace trust
3) The schema still won't download, and will show a message for the download being disabled
4) If you reload the window it should work as expected again

This is because workspace trust is checked once at activation, but we're not using the `onDidGrantWorkspaceTrust` event. See https://github.com/redhat-developer/vscode-java/blob/master/src/extension.ts#L447-L449 for how to do it. I guess this may also be an issue with resolving entities ?

**Review by rgrunber — APPROVED** [review]
I'm fine with merging. @AlexXuChen , can address the issue I mentioned in a separate PR.

**Comment by fbricon** [comment]
thanks @angelozerr!

**Pull request merged**